File: iteration2/src/Advisor.java
import java.util.ArrayList;

class Advisor extends Lecturer {

    private ArrayList<Student> students = new ArrayList<Student>();

    public Advisor(String name, String surname, String email, String phoneNumber, String ID, String password, String faculty, String department, ArrayList<Course> givenCourses, String academicTitle) {
        super(name, surname, email, phoneNumber, ID, password, faculty, department, givenCourses, academicTitle);
    }


    public ArrayList<Student> getStudents(){ 
        return students;
    }

    public void setStudent(Student student){
        this.students.add(student);
    }
}

File: iteration3/src/advisor.py
from lecturer import Lecturer
class Advisor(Lecturer):
    def __init__(self, name, surname, email, phone_number, id, password, faculty, department, academic_title):
        super().__init__(name, surname, email, phone_number, id, password, faculty, department, academic_title)
        self.__students = []

    def set_student(self, student):
        self.__students.append(student)
    
    def get_students(self):
        return self.__students